internal/pfcpsim/helpers.go

```diff
@@ -47,38 +43,33 @@ func isRemotePeerConnected() bool {
 	return remotePeerConnected
 }
 
-// getLocalAddress retrieves local address in string format to use when establishing a connection with PFCP agent
-func getLocalAddress() (string, error) {
-	// cmd to run for darwin platforms
-	cmd := "route -n get default | grep 'interface:' | grep -o '[^ ]*$'"
-
-	if runtime.GOOS != "darwin" {
-		// assuming linux platform
-		cmd = "route | grep '^default' | grep -o '[^ ]*$'"
-	}
-
-	cmdOutput, err := exec.Command("bash", "-c", cmd).Output()
+// getLocalAddress retrieves the IP of the first non-loopback interface.
+// Returns IP address of interfaceName, if specified.
+// Returns error if fail occurs at any stage.
```

Comment:
Comment seems wrong, from my understanding: 
```suggestion
// getLocalAddress returns the first IP address of the interfaceName, if specified,
// otherwise returns the IP address of the first non-loopback interface.
// Returns error if fail occurs at any stage.
```